File: ClientApp/src/app/core/models/full-screen-form-editor/full-screen-form-model.ts
import { ControlType } from "../controls/control-type";
import { FormValueModel } from "../form-editor/form-value-model";
import { TileCode } from "../tile-code";

export interface FullScreenFormModel {
    formValueModel?: FormValueModel;
    tileCode: TileCode;
    controls?: ControlPreviewModel[];
    selectedControl?: string;
}

export interface ControlPreviewModel {
    id: string;
    type: ControlType;
}

File: FinanceTracker/Models/FullScreenFormModel.cs
using FinanceTracker.Core.Models;
using FinanceTracker.Core.Models.Controls;
using FinanceTracker.Core.Models.Forms;
using FinanceTracker.Core.Models.FullScreenModels;

namespace FinanceTracker.Models
{
    public class FullScreenFormModel
    {
        public FormValueModel? FormValueModel { get; set; } = new();
        public TileCode TileCode { get; set; }
        public List<FormComponent> Controls { get; set; } = [];
        public string? SelectedControl { get; set; }
    }
}
